ads/google/a4a/utils.js

```diff
@@ -293,3 +310,83 @@ export function getCorrelator(win, opt_cid) {
   }
   return win.ampAdPageCorrelator;
 }
+
+/**
+ * Gets a relatively terse and controlled-format string version of the user
+ * agent.
+ *
+ * @param {!Window} win
+ * @return {string}
+ */
+export function userAgentString(win) {
+  const platform = platformFor(win);
+  let browser;
+  let os;
+  let version;
+  if (platform.isIos()) {
+    os = 'ios';
+  } else if (platform.isAndroid()) {
+    os = 'android';
+  } else {
+    os = 'other_os';
+  }
+  if (platform.isSafari()) {
+    browser = 'safari';
+  } else if (platform.isChrome()) {
+    browser = 'chrome';
+  } else {
+    browser = 'other_browser';
+  }
+  version = platform.getMajorVersion();
+  return `${browser}_${version}_on_${os}`;
+}
+
+/**
+ * Creates or reinitializes a lifecycle reporter for Google ad network
+ * implementations.
+ *
+ * @param {!../../../extensions/amp-a4a/0.1/amp-a4a.AmpA4A} a4aElement
+ * @return {!./performance.GoogleAdLifecycleReporter}
+ */
+export function googleLifecycleReporterFactory(a4aElement) {
+  const reporter =
+      /** @type {!./performance.GoogleAdLifecycleReporter} */
+      (getLifecycleReporter(a4aElement, 'a4a', undefined,
+          a4aElement.element.getAttribute('data-amp-slot-index')));
+  const slotId = reporter.getSlotId();
+  reporter.setPingVariable('v_h', 'VIEWPORT_HEIGHT');
```

Comment:
True, but at the moment there's only one place that has to set more than one variable.  Good generalization for the future, when we have more variables being inserted, though.